src/Workspaces/Core/Portable/Workspace/Host/TextFactory/ITextFactoryService.cs

```diff
@@ -16,6 +16,8 @@ namespace Microsoft.CodeAnalysis.Host
     /// </summary>
     internal interface ITextFactoryService : IWorkspaceService
     {
+        SourceHashAlgorithm ChecksumAlgorithm { get; }
```

Comment:
> SourceHashAlgorithm ChecksumAlgorithm { get; } [](start = 7, length = 47)

Not sure what is this good for? All implementations return SHA1.

Reply:
It ensures that the use of this property will not be missed if the implementation changes in the future.

Reply:
ITextFactoryService doesn't seem like the right place for this. The service is for creating buffers.

Reply:
This property returns the `ChecksumAlgorithm` used for `SourceText` instances created by this service. Since this service creates the `SourceText` instances, it is the only one that can know what checksum algorithm will be used.

Reply:
It would be imo better to pass the alg as an argument to the methods of this interface.

Reply:
➡️ Property removed by using the value from serialization.